app/assets/javascripts/oxalis/model_initialization.js

```diff
@@ -98,14 +97,14 @@ export async function initialize(
 
     Store.dispatch(setTaskAction(annotation.task));
   } else {
-    // In View mode, the annotationId is actually the datasetName
-    // as there is no annotation and no tracing!
-    datasetName = annotationIdOrDatasetName;
+    // In View Mode the TraceOrViewCommandType contains the dataset name and the owning organization
```

Comment:
Thanks to your refactoring, you can remove this comment, I think :) Previously, it was very weird that annotationId equaled datasetName. Now, it should be self-explanatory when looking at the types.